src/ccnl-core/src/ccnl-prefix.c

```diff
@@ -578,62 +578,49 @@ ccnl_prefix_to_path_detailed(struct ccnl_prefix_s *pr, int ccntlv_skip,
     else
         buf = prefix_buf2;
     */
-    char *buf = (char*) ccnl_malloc(CCNL_MAX_PREFIX_SIZE);
+    char *buf = (char*) ccnl_malloc(CCNL_MAX_PREFIX_SIZE+1);
     if (buf == NULL) {
         DEBUGMSG_CUTL(ERROR, "ccnl_prefix_to_path_detailed: malloc failed, exiting\n");
         return NULL;
     }
-
+    memset(buf, 0, CCNL_MAX_PREFIX_SIZE+1);
     return ccnl_prefix_to_str_detailed(pr, ccntlv_skip, escape_components, call_slash, buf, CCNL_MAX_PREFIX_SIZE);
 }
 
 char*
 ccnl_prefix_to_str_detailed(struct ccnl_prefix_s *pr, int ccntlv_skip, int escape_components, int call_slash,
-                            char *buf, int buflen) {
-    int len = 0, i, j;
+                            char *buf, size_t buflen) {
+    size_t len = 0, i, j;
     (void)i;
     (void)j;
     (void)len;
     (void) call_slash;
-    (void)buflen; //fixme, prevent bufferoverrun here!
 #ifdef USE_NFN
-    // len += sprintf(buf + len, "cmpcnt: %i|", pr->compcnt);
     if (pr->nfnflags & CCNL_PREFIX_NFN) {
-        len += sprintf(buf + len, "nfn");
+        len += snprintf(buf + len, buflen - len, "nfn");
+        if (buflen < len){
+            return NULL;
+        }
     }
 #ifdef USE_NFN_REQUESTS
-    // len += sprintf(buf + len, ":test");
     if (pr->nfnflags & CCNL_PREFIX_REQUEST) {
         char *desc = nfn_request_description_new(pr->request);
-        len += sprintf(buf + len, ":%s", desc);
+        len += snprintf(buf + len, buflen - len, ":%s", desc);
+        if (buflen < len){
+            return NULL;
+        }
         ccnl_free(desc);
-        // if (pr->request == NULL) {
-        //     len += sprintf(buf + len, ":request(?)");
-        // } else {
-        //     len += sprintf(buf + len, ":request(%d)", pr->request->complen);
-        // }
-        // len += sprintf(buf + len, ":request(%c)", *(pr->request->comp));
-        // len += sprintf(buf + len, ":request(%.*s)", pr->request->complen, pr->request->comp);
+
     }
 #endif // USE_NFN_REQUESTS
     if (pr->nfnflags) {
-        len += sprintf(buf + len, "[");
+        len += snprintf(buf + len, buflen - len, "[");
```

Comment:
Are the use flags in this function exclusive, meaning only one is executed at times? If not, and other branches are executed as well, than the ``len +=`` overwrites any potential errors.